pkg/ingestor/parser/spdx/parse_spdx.go

```diff
@@ -263,6 +263,9 @@ func (s *spdxParser) GetPredicates(ctx context.Context) *assembler.IngestPredica
 		} else if isDependent(rel.Relationship) {
 			foundId = string(rel.RefB.ElementRefID)
 			relatedId = string(rel.RefA.ElementRefID)
+		} else if isPackageOf(rel.Relationship) {
```

Comment:
A suggestion to avoid code duplication. Could this be refactored into something like this?
```golang
func getIds(rel RelationshipType) (string, string) {
    var foundId, relatedId string

    if isDependency(rel.Relationship) || isDependent(rel.Relationship) || isPackageOf(rel.Relationship) {
        foundId = string(rel.RefB.ElementRefID)
        relatedId = string(rel.RefA.ElementRefID)
    }

    return foundId, relatedId
}
```
``` golang
foundId, relatedId := getIds(rel)
if foundId == "" || relatedId == "" {
    continue
}
```

Reply:
Thanks for the review @naveensrinivasan  - I've altered the code so there isn't a separate `else if` block just for the new `isPackageOf` condition but we need a separate shared block for that and `isDependent` because the ids have to be the other way round for those compared to `isDependency`.